Peaky/IParametrizedTestCases.cs

```diff
@@ -0,0 +1,7 @@
+namespace Peaky
+{
+    public interface IParametrizedTestCases : IPeakyTest
```

Comment:
I suspect we can do without a public interface for this. This feels like an internal implementation detail.

Reply:
how do we then expose the registration call back?